test/WebJobs.Script.Tests/Rpc/WorkerConfigFactoryTests.cs

```diff
@@ -207,5 +216,70 @@ public void AddArgumentsFromAppSettings_JavaOpts(string expectedArgument, string
                 Assert.Equal(expectedArgument, workerDescription.Arguments[1]);
             }
         }
+
+        [Theory]
+        [InlineData("{FUNCTIONS_WORKER_RUNTIME_VERSION}/@os/@architecture", "3.7/LINUX/X64")]
+        [InlineData("{FUNCTIONS_WORKER_RUNTIME_VERSION}/@architecture", "3.7/X64")]
+        [InlineData("{FUNCTIONS_WORKER_RUNTIME_VERSION}/@os", "3.7/LINUX")]
+        public void LanguageWorker_HydratedWorkerPath_EnvironmentVersionSet(string defaultWorkerPath, string expectedPath)
+        {
+            Environment.SetEnvironmentVariable(LanguageWorkerConstants.FunctionWorkerRuntimeVersionSettingName, "3.7");
```

Comment:
you need to use TestScopedSettings to set env variables for tests as it will ensure the environment is cleaned up after tests

Reply:
I'm not sure this is needed given that we are going to be passing the `IEnvironment` to the `WorkerConfigFactory` as an argument. I've modified the `WorkerConfigFactoryTests` so that they provide a clean test environment every time the tests are run.